src/misc/VarNames.jl

```diff
@@ -0,0 +1,488 @@
+import MacroTools as MT
+import Printf
+
+@assert VarName === Union{Symbol, AbstractString, Char}
+const VarNames = Union{
+    VarName,
+    AbstractArray{<:VarName},
+    Pair{<:VarName},
+}
+
+req(cond, msg) = cond || throw(ArgumentError(msg))
+# The macro version will only evaluate `msg` if needed
+macro req(cond, msg) :($cond || throw(ArgumentError($msg))) end
```

Comment:
I am slightly puzzled about this: you are not actually using `@req`, just calling the function `req`?

In Hecke we define and export the following (I omitted the docstring):
```
macro req(cond, msg)
  quote
    if !($(esc(cond)))
      throw(ArgumentError($(esc(msg))))
    end
  end
end
```
Except for the `esc` around the message I think your implementation is equivalent?

I'd be happy to transfer the `@req` macro to AA anyway, it would make sense. For that matter, probably everything in Hecke's `src/Assertions.jl` could be moved... But not in *this* PR of course!

Reply:
I used `@req` in some cases before, where `msg` depended on `cond` being false. Indeed it should have the `esc`. Anyway, I opened https://github.com/thofma/Hecke.jl/issues/1264 for moving this to AbstractAlgebra and would then replace all occurrences of `req` by `@req`.